### PR TITLE
[REF] [Export] Simplify setting of address strings

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor refactor of export strings (this is well tested code)

Before
----------------------------------------
Less readable

After
----------------------------------------
More readable

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

**Comment by colemanw** [comment]
Hmm, I don't quite understand how this simplifies things. It's an additional param to `ExportProcessor::__costruct` and appears to increase coupling with the form without reducing code in the form.

**Comment by eileenmcnaughton** [comment]
@colemanw well $exportParams is ALREADY the formValues array - so we are renaming it to be more accurate. In addition we are moving it out of random place to being analysed at the start so we can stop passing the array around the exportComponents function. I'd rather split it out better but this gets it out of the 'deep down' part of the function

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
test this please

File: CRM/Export/BAO/ExportProcessor.php [hunk 1]
```diff
@@ -378,8 +378,14 @@ public function setAddresseeGreetingTemplate($addresseeGreetingTemplate) {
...
-   */
-  public function __construct($exportMode, $requestedFields, $queryOperator, $isMergeSameHousehold = FALSE, $isPostalableOnly = FALSE, $isMergeSameAddress = FALSE) {
+   * @param array $formValues
+   *   Values from the export options form on contact export. We currently support these keys
+   *   - postal_greeting
+   *   - postal_other
+   *   - addresee_greeting
+   *   - addressee_other
+   */
+  public function __construct($exportMode, $requestedFields, $queryOperator, $isMergeSameHousehold = FALSE, $isPostalableOnly = FALSE, $isMergeSameAddress = FALSE, $formValues = []) {
```
Comment (colemanw) [inline CRM/Export/BAO/ExportProcessor.php]: We haven't adopted an official documentation standard for arrays, but I like the looks of this one: https://stackoverflow.com/a/51032174/2918050

**Comment by colemanw** [comment]
Ok I'll take your word for it. I left one comment about the formatting of the docblock, but other than that it looks good.

**Pull request merged**